osquery/filesystem/linux/proc.h

```diff
@@ -159,10 +167,17 @@ Status procEnumerateProcesses(UserData& user_data,
       if (ret == false) {
         break;
       }
+
+      anySuccess = true;
+    } catch (const boost::filesystem::filesystem_error& e) {
+      VLOG(1) << "Exception iterating Linux processes: " << e.what()
+              << "Iterating onward\n";
     }
-  } catch (const boost::filesystem::filesystem_error& e) {
-    VLOG(1) << "Exception iterating Linux processes: " << e.what();
-    return Status(1, e.what());
+  }
+
+  if (!anySuccess) {
+    VLOG(1) << "No success iterating linux /proc. Permissions problems?\n";
+    return Status(1, "No success iterating linux /proc");
```

Comment:
Prefer the new `Success::failure("No success iterating linux /proc");`

Reply:
👍